middlewares/ratingValidation.js

```diff
@@ -0,0 +1,44 @@
+import { isNumeric } from 'validator';
+
+export default {
+  verifyMentor(req, res, next) {
+    const { isMentor } = req.decoded;
+    if (isMentor === false) {
+      const error = new Error('Only mentors can rate articles');
+      error.status = 401;
+      return next(error);
+    }
+    return next();
+  },
+
+  validateInput(req, res, next) {
+    const { rateValue } = req.body;
+    if (!(rateValue)) {
```

Comment:
The if block will be executed for `rateValue: ''`. I suppose that what you are trying to do here is check for the existence of the `rateValue` key in the request body. As such, more useful options could be:

`if (!Object.keys(req.body).includes(rateValue))` or `if (!(rateValue in req.body))`

Reply:
Also, you can add one more middleware for checking the trimmed value of `rating` and send an error message. Although technically, your check for type with `isNumeric` should cover that since an empty string or a string with an infinite number of spaces cannot be type cast into a number.